src/main/java/us/ihmc/commons/thread/RepeatingTaskThread.java

```diff
@@ -343,17 +343,14 @@ public final void run()
          // If a period/frequency limit was set, wait until loop can run.
          if (periodLowerLimit > 0.0)
          {
-            /*
-             * This call must not swallow interrupts.
-             * As of writing this comment (Nov, 2024), LockSupport.parkNanos() is used internally to block.
-             * Although the throttler will block until the period has elapsed, the thread
-             * remains interrupted.
-             */
+            // Uses ThreadTools.parkAtLeast, which will not throw InterruptedException, but
+            // will return early when this thread is interrupted.
```

Comment:
I think this is false, because parkAtLeast loops until the desired time has elapsed. It might just free spin if parkNanos is returning immediately due to the interrupt. We should probably fix that haha.

Reply:
I fixed this in https://github.com/ihmcrobotics/ihmc-commons/pull/41